87.md

```diff
@@ -0,0 +1,129 @@
+NIP-87
+======
+
+Ecash Mint Discoverability
+--------------------------------
+
+`draft` `optional`
+
+This NIP describes `kind:38173`, `kind:38172` and `kind:18173`: a way to discover ecash mints, their capabilities, and people who recommend them.
+
+## Rationale
+
+Nostr's discoverability and transparent event interaction is one of its most interesting/novel mechanics.
+This NIP provides a simple way for users to discover ecash mints recommended by other users and to interact with them.
+
+### Parties involved
+
+There are three actors to this workflow:
+
+* An ecash mint operator, announces their mint and its capabilities.
+    * Publishes `kind:38173` or `kind:38172`, detailing how to connect to it and its capabilities.
+* user A, who recommends an ecash mint
+    * Publishes `kind:18173`
+* user B, who seeks a recommendation for an ecash mint
+    * Queries for `kind:18173` and, based on results, queries for `kind:38173`/`kind:38172`
+
+## Events
+
+### Recommendation event
+```json
+{
+  "kind": 18173,
+  "pubkey": <recommender-user-pubkey>,
+  "tags": [
+    ["u", <recommended-cashu-mint-url>, "cashu"],
+    ["u", <recommended-fedimint-invite-code>, "fedimint"],
+    ["u", <recommended-cashu-mint-url>, "cashu"],
+    ["a", "38173:fedimint-pubkey:<d-identifier>", "wss://relay1", "fedimint"],
```

Comment:
Why multiple u tags for one recommendation?
Can it be one u tag per recommendation and then include metadata like a review and rating, or anything else to help clients rank mints? Would also make handling the events simpler

Reply:
It is a list, so you include all your recommendations in one event.

review and ratings don't really make sense for this use case, you either trust them with your money or you don't/

Reply:
I think we need comments and reviews because there will be several mints that I like/use but I rate them based on stars and put my comment for what I like or what I use. NIP-89 (nostrapp.link) uses kind 1985 .  See the sample event below. We should use the same format for ratings and recommendations. 

{
  "created_at": 1710460808,
  "content": "The best text 2 speech I;ve used.. ",
  "tags": [
    [
      "l",
      "review/app",
      "social.coracle.ontology",
      "{\"quality\":1}"
    ],
    [
      "L",
      "social.coracle.ontology"
    ],
    [
      "p",
      "cf13a4f45628931f1cdbe7d31f923b6f6884d41644ab2cc055a0e0805b77cb9e"
    ],
    [
      "a",
      "31990:cf13a4f45628931f1cdbe7d31f923b6f6884d41644ab2cc055a0e0805b77cb9e:D94C72C241D1C73B2C36EEA73E9FB"
    ]
  ],
  "kind": 1985,
  "pubkey": "d8a2c33f2e2ff3a9d4ff2a5593f3d5a59e9167fa5ded063d0e49891776611e0c",
  "id": "95f5d58a7c9c2caed4f77d403c6e1846c71644997c87ed112f569413ec13bf9b"
}

Reply:
> review and ratings don't really make sense for this use case, you either trust them with your money or you don't/

A mint could have lots of downtime, poor lighting routing, slow responses etc, but still be trusted to hold my money

Reply:
@benthecarman I propose the following changes to recommendation event kind 18173. 

Add content tag to provide verbal comments on the recommendation.
Add 'l' tag include review stars like this. Quality can range from 0 to 1. 

{"l","<type>", "{"quality":1}"}

Reply:
Oh god, please don't 😔

Reply:
I would propose something more like:

Content field is for verbal comments/reviews. 
Add a tag like "rating" that could be 0 to 100: ["rating", 42]... no need to index the rating and I like whole numbers. 

Cc @starbackr-dev @benthecarman

Reply:
Use the reviews NIP: https://github.com/nostr-protocol/nips/pull/879/files

Reply:
Yes. Kind 1986 will exactly fit this. Let's use that for recommendation, comments or review and use 18173 for mint information.

Reply:
How do I then signal that a mint is untrusted? There needs to be a way to explicitly say that certain services are bad. Either a new kind for *dissuasion* or the *recommendation* needs to be generalized to *advice* with positive or negative feedback.
I see that ratings may be overkill because as @benthecarman said, "you either trust them with your money, or you don't", but we should know which mints are **not** trusted with our money.

Reply:
Leave a bad review?

Reply:
And you propose using nip85 separately to do that? Nip87 for positive recommendations and nip85 for negative reviews?

Reply:
Added a line for the recommendation events to have the content be a review. This should suffice